mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/AssemblyMirror.cs

```diff
@@ -128,5 +141,68 @@ public TypeMirror GetType (String name, Boolean throwOnError)
 				meta = value;
 			}
 		}
+		
+		public byte[] GetMetadataBlob() {
+			if (metadata_blob != null)
+				return metadata_blob;
+			
+			vm.CheckProtocolVersion(2, 47);
+
+			return metadata_blob = vm.conn.Assembly_GetMetadataBlob(id);
+		}
+		
+		public bool IsDynamic {
+			get {
+				if (isDynamic.HasValue)
+					return isDynamic.Value;
+				
+				vm.CheckProtocolVersion(2, 47);
+
+				isDynamic = vm.conn.Assembly_IsDynamic(id);
+				return isDynamic.Value;
+			}
+		}	
+		
+		public bool HasPdb {
+			get {			
+				return GetPdbBlob().Length > 0;
+			}
```

Comment:
A method named HasSomething probably shouldn't load mb-s of data from the network.